library/core/src/slice/iter.rs

```diff
@@ -458,8 +458,12 @@ where
         match self.v.iter().position(|x| (self.pred)(x)) {
             None => self.finish(),
             Some(idx) => {
-                let ret = Some(&self.v[..idx]);
-                self.v = &self.v[idx + 1..];
+                let (left, right) =
+                    // SAFETY: if v.iter().position returns Some(idx), that
+                    // idx is definitely a valid index for v
+                    unsafe { (self.v.get_unchecked(..idx), self.v.get_unchecked(idx + 1..)) };
```

Comment:
does `v.split_at_unchecked` work here?

Reply:
No, `split_at_unchecked` would be equivalent to `(self.v.get_unchecked(..idx), self.v.get_unchecked(idx..)`